src/tracing/esp32_trace/counter.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <esp_diagnostics_metrics.h>
+#include <esp_log.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CHIPMemString.h>
+
+#define PATH "insights.cnt"
+
+namespace Insights {
+class ESPInsightsCounter
+{
+private:
+    static ESPInsightsCounter * mHead; // head of the counter list
+    char label[50];                    // unique key
```

Comment:
my previous comment was lost due to force push.
Re-adding here to be up to date: why is this expected to be dynamic and in what way? I only see static strings so far. Could you give examples?

Asking because for other tracing mechanisms we explicitly require static strings because that allows tracing to be less intrusive.

Reply:
I have addressed this in my latest commit.